### PR TITLE
Add support for multiple aliases

### PR DESCRIPTION
**Pull request by shadowspawn**
# Pull Request

See #531

## Problem

Some people would like to add more than one alias for a command, but not have them all visible in the help.

## Solution

Allow adding multiple aliases. Only display the first one in the built-in help, and leave any mention of the rest up to the caller.

This is intended to be backwards compatible. There is a small change from v5.0 behaviour, calling `.alias()` when there is no alias will now return `undefined` instead of `null`, but seems unlikely to affect much code and can be made fully backwards compatible if necessary.

To provide public access to the possibly multiple aliases, added `.aliases()` which works with array rather than string.

## ChangeLog

- add support for multiple aliases

### CODE REVIEW

**Comment by kodie** [comment]
This is awesome!

File: typings/index.d.ts [hunk 1]
```diff
@@ -275,6 +275,8 @@ declare namespace commander {
     /**
      * Set an alias for the command.
      *
+     * You may call more than once to add multiple aliases, Only the first alias is shown in the auto-generated help.
```
Comment (victornpb) [inline typings/index.d.ts:278]:
```suggestion
     * You may call more than once to add multiple aliases, only the first alias is shown in the auto-generated help.
```
just catched a small typo, a capitalized word after a comma.

File: index.js [hunk 2]
```diff
@@ -1231,7 +1233,25 @@ class Command extends EventEmitter {
...
+   *
+   * @param {string[]} [aliases]
+   * @return {string[]|Command}
+   * @api public
+   */
+
+  aliases(aliases) {
+    // Getter for the array of aliases is the main reason for having aliases() in addition to alias().
+    if (aliases === undefined) return this._aliases;
+
```
Comment (victornpb) [inline index.js:1253]: What do you think about this?
```suggestion
    if (arguments.length > 1) aliases = Array.from(arguments);
```

**Comment by shadowspawn** [comment]
Changes: added `.aliases()` mainly to give public access to full list of aliases (e.g. for code adding command-line completion).

File: index.js (hunk 2 above)
Comment (shadowspawn) [reply index.js:1253]: Fair question. Using `arguments.length` means some smart editors list multiple arguments in the auto-detected signature. I actually removed all uses just recently! #1235

An alternative would to use separate arguments _instead_ of array. I prefer array in this case for the symmetry with the getter/setter signatures.

Comment (victornpb) [reply index.js:1253]: Oh ok, I don't even know what would be the ts definition for that.
Array makes more sense from a typing pov, and varargs is just a preferred lazy typing person.

**Comment by shadowspawn** [comment]
I'll leave PR in draft for a few more days for any other feedback. 

Thanks @kodie and @victornpb and @abetomo

**Comment by shadowspawn** [comment]
No further feedback. Moving out of draft. 

I have deliberately not added to README for now, hopefully anyone who wants the feature will be able to find it easily enough, and the JSDoc and TSDoc cover the support for multiple aliases.

**Comment by shadowspawn** [comment]
Included in Commander v5.1 which has been released.

https://github.com/tj/commander.js/releases/tag/v5.1.0

**Comment by viceice** [comment]
This is blocking us from upgrading, since we use `aliasses` as an options.

Wha'ts the best compatible way to solve this.

**Comment by shadowspawn** [comment]
Sorry for the clash @viceice 

See the section in the README about avoiding name clashes, and have a look at `storeOptionsAsProperties` and `passCommandToAction`:
- https://github.com/tj/commander.js#avoiding-option-name-clashes

**Comment by viceice** [comment]
@shadowspawn Ok, thanks for the info. Works with the suggested changes.

**Pull request merged**